arrow-data-source/standard/src/main/scala/com/intel/oap/spark/sql/execution/datasources/v2/arrow/ArrowUtils.scala

```diff
@@ -113,8 +116,42 @@ object ArrowUtils {
       input.close()
     }
 
+    val totalVectors = if (nullVectors.nonEmpty) {
+      val finalVectors =
+        mutable.ArrayBuffer[ArrowWritableColumnVector]()
+      val requiredIterator = requiredSchema.iterator
+      val caseSensitive = SQLConf.get.caseSensitiveAnalysis
+      while (requiredIterator.hasNext) {
+        val field = requiredIterator.next()
+        finalVectors.append(
+          if (caseSensitive) {
+            vectors.find(_.getValueVector.getName.equals(field.name))
+              .getOrElse {
+                // The missing column need to be find in nullVectors
+                val nullVector = nullVectors.find(_.getValueVector.getName.equals(field.name)).get
+                nullVector.setValueCount(rowCount)
+                nullVector.retain()
+                nullVector
+              }
+          } else {
+            vectors.find(_.getValueVector.getName.equalsIgnoreCase(field.name))
+              .getOrElse {
+                // The missing column need to be find in nullVectors
+                val nullVector =
+                  nullVectors.find(_.getValueVector.getName.equalsIgnoreCase(field.name)).get
```

Comment:
Just a small suggestion for code refinement. I think the different handling for case sensitive/insensitive can be simplified. We can define a lambda expression as below. Thus, we can directly use this `eql` in finding matched field, like `anArray.find(x => eql(x, "MATCH_TARGET"))`, without separating the code logic with if/else. We can do the similar code refinement in other places. Right?
```
    val eql = if (caseSensitive) {
      (a: String, b: String) => a.equals(b)
    } else {
      (a: String, b: String) => a.equalsIgnoreCase(b)
    }
```